bai2/main.tf

```
@@ -0,0 +1,338 @@
+
+  egress {
+    from_port        = 0
+    to_port          = 0
+    protocol         = "-1"
+    cidr_blocks      = ["0.0.0.0/0"]
+    ipv6_cidr_blocks = ["::/0"]
+  }
+
+  tags = {
+    Name = "app-demo-sg"
+  }
+}
+
+resource "aws_security_group_rule" "example11" {
+  type              = "ingress"
+  from_port         = 22
+  to_port           = 22
+  protocol          = "tcp"
+  source_security_group_id = aws_security_group.basion-demo-sg.id 
+  security_group_id = aws_security_group.app-demo-sg.id
+}
+
+resource "aws_security_group_rule" "example" {
+  type              = "ingress"
+  from_port         = 22
+  to_port           = 22
+  protocol          = "tcp"
+  cidr_blocks       = ["0.0.0.0/0"]
+  security_group_id = aws_security_group.basion-demo-sg.id
+}
+
+#security group cho alb
+resource "aws_security_group" "alb-demo-sg" {
+  name        = "lb-demo-sg"
+  description = "Allow ssh inbound traffic"
+  vpc_id      = aws_vpc.vpc-demo-1.id
+
+  ingress {
+    description      = "HTTP"
+    from_port        = 80
+    to_port          = 80
+    protocol         = "tcp"
+    cidr_blocks      = ["0.0.0.0/0"]
+    
+  }
+
+  ingress {
+    description      = "SSH"
+    from_port        = 22
+    to_port          = 22
+    protocol         = "tcp"
+    cidr_blocks      = ["0.0.0.0/0"]
+    
+  }
+
+  egress {
+    from_port        = 0
+    to_port          = 0
+    protocol         = "-1"
+    cidr_blocks      = ["0.0.0.0/0"]
+    ipv6_cidr_blocks = ["::/0"]
+  }
+
+  tags = {
+    Name = "alb-demo-sg"
+  }
+}
+
+# tao network interface
+resource "aws_network_interface" "test1" {
+  subnet_id       = aws_subnet.subnet-public-1.id
+  private_ips     = ["10.0.1.50"]
+  security_groups = [aws_security_group.basion-demo-sg.id]
+
+}
+
+resource "aws_network_interface" "test2" {
+  subnet_id       = aws_subnet.subnet-public-2.id
+  private_ips     = ["10.0.4.51"]
+  security_groups = [aws_security_group.app-demo-sg.id]
+}
+
+#tao ElasticIP
+ resource "aws_eip" "one" {
+  vpc                       = true
+  network_interface         = aws_network_interface.test1.id
+  associate_with_private_ip = "10.0.1.50"
+  depends_on                = [aws_internet_gateway.gw-demo]
+}
+
+resource "aws_eip" "two" {
+  vpc                       = true
+  network_interface         = aws_network_interface.test2.id
+  associate_with_private_ip = "10.0.4.51"
+  depends_on                = [aws_internet_gateway.gw-demo]
+}
+
+#tao ec2 instance
+resource "aws_instance" "basion-demo-ec2" {
+  ami           = "ami-055d15d9cfddf7bd3"
+  instance_type = "t2.micro"
+  key_name = "bastion-key"
+
+  network_interface {
+    network_interface_id = aws_network_interface.test1.id
+    device_index         = 0
+  }
+  
+  credit_specification {
+    cpu_credits = "unlimited"
+  }
+  
+  tags = {
+      name = "basion-demo-ec2"
+  }
+}
+
+resource "aws_instance" "app-demo-ec2" {
+  ami           = "ami-055d15d9cfddf7bd3"
+  instance_type = "t2.micro"
+  key_name = "app-key-1"
+
```

```suggestion
iam_instance_profile = aws_iam_role.ec2-access-s3

```
gán role vừa tạo được vào EC2